databases/week3/databases_week3_mealSharing_DDL.sql

```diff
@@ -0,0 +1,222 @@
+CREATE DATABASE databases_week3_mealsharing DEFAULT CHARACTER SET = 'utf8mb4';
+USE databases_week3_mealsharing;
+CREATE TABLE `meal` (
+  `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `title` VARCHAR(255) NOT NULL,
+  `description` TEXT NOT NULL,
+  `location` VARCHAR(255) NULL,
+  `when` DATETIME NOT NULL,
+  `max_reservations` INT(10) NOT NULL,
+  `price` DECIMAL(7, 2) NOT NULL,
+  `created_date` DATE NOT NULL
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `reservation` (
+  `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `number_of_guests` INT(10) NOT NULL,
+  `created_date` DATE NOT NULL,
+  `contact_phonenumber` VARCHAR(255) NOT NULL,
+  `contact_name` VARCHAR(255) NOT NULL,
+  `contact_email` VARCHAR(255) NOT NULL,
+  `meal_id` INT(10) UNSIGNED NOT NULL,
+  CONSTRAINT `fk_reservation_meal` FOREIGN KEY (`meal_id`) REFERENCES `meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
If we delete all reservations for a given meal, the meal entry will also be deleted, is that what you want? (It is ok, just checking )

Reply:
it gave me a error while i wrote ON DELETE NULL ON UPDATE CASCADE so i just keep like that. but i understand the situation that it delets all the entries refering to that id . 
was that not correct writting that syntax with NULL ?

Reply:
it should ON DELETE SET NULL ON UPDATE CASCADE !! 

did that thanks